packages/runner/src/runner.ts

```diff
@@ -1081,4 +1131,43 @@ export class Runner {
     this.logger("error", errMsg);
     throw new Error(errMsg);
   }
+
+  async _injectStakeCell(
+    txSkeleton: TransactionSkeletonType
+  ): Promise<TransactionSkeletonType> {
+    // Add stake lock dep
+    txSkeleton = txSkeleton.update("cellDeps", (cellDeps) => {
+      return cellDeps.push(this.config.deploymentConfig.stake_lock_dep);
+    });
+
+    const oldStakeCell: Cell = await this._queryValidStakeCell();
+    // Add stake cell input
+    txSkeleton = txSkeleton.update("inputs", (inputs) =>
+      inputs.push(oldStakeCell)
+    );
+    // Add stake cell output
+    let newStakeCell: Cell = oldStakeCell;
+    const oldStakeLockArgs = this._unpackStakeLockArgs(
+      oldStakeCell.cell_output.lock!.args
+    );
+    // Update stake_block_number to the current L2 block number
+    if (BigInt(oldStakeLockArgs.stake_block_number) === BigInt(0)) {
```

Comment:
The args should always be updated when submitting a block.

Reply:
fixed.